lib/graphql/railtie.rb

```diff
@@ -1,12 +1,25 @@
 # frozen_string_literal: true
+
 module GraphQL
   class Railtie < Rails::Railtie
-    config.before_configuration do
-      # Bootsnap compile cache has similar expiration properties,
-      # so we assume that if the user has bootsnap setup it's ok
-      # to piggy back on it.
-      if ::Object.const_defined?("Bootsnap::CompileCache::ISeq") && Bootsnap::CompileCache::ISeq.cache_dir
-        Language::Parser.cache ||= Language::Cache.new(Pathname.new(Bootsnap::CompileCache::ISeq.cache_dir).join('graphql'))
+    config.graphql = ActiveSupport::OrderedOptions.new
+    config.graphql.parser_cache = false
+
+    config.before_initialize do
+      if !config.graphql.parser_cache && ::Object.const_defined?("Bootsnap::CompileCache::ISeq") && Bootsnap::CompileCache::ISeq.cache_dir
+        Deprecation.warn <<~MSG.squish
+          The GraphQL parser cache must be explicitly enabled in your application.
+          Please add `config.graphql.parser_cache = true` to your application config.
+        MSG
+        config.graphql.parser_cache = true
+      end
```

Comment:
IMO this is too much.

Reply:
I assume you mean the deprecation warning specifically is too much? Do you think the behaviour change would be obvious enough by reading release notes, or do you have a different idea?

Reply:
I mean the whole if. Changelog should be enough. This cache is a nice to have, but I wouldn't call it a breaking change if it's no longer configured by default.

Reply:
👍 If @rmosolgo agrees I'll squash commits and we can proceed.